packages/core/src/application.ts

```diff
@@ -293,125 +50,118 @@ export class Application extends Context {
   }
 
   /**
-   * Register a new Controller-based route.
+   * Bind a Server constructor to the Application's master context.
+   * Each server constructor added in this way must provide a unique prefix
+   * to prevent binding overlap.
    *
    * ```ts
-   * class MyController {
-   *   greet(name: string) {
-   *     return `hello ${name}`;
-   *   }
-   * }
-   * app.route('get', '/greet', operationSpec, MyController, 'greet');
+   * app.server(RestServer);
+   * // This server constructor will be bound under "servers.RestServer".
+   * app.server(RestServer, "v1API");
+   * // This server instance will be bound under "servers.v1API".
    * ```
    *
-   * @param verb HTTP verb of the endpoint
-   * @param path URL path of the endpoint
-   * @param spec The OpenAPI spec describing the endpoint (operation)
-   * @param controller Controller constructor
-   * @param methodName The name of the controller method
+   * @param {Constructor<Server>} server The server constructor.
+   * @param {string=} name Optional override for key name.
+   * @memberof Application
    */
-  route(
-    verb: string,
-    path: string,
-    spec: OperationObject,
-    controller: ControllerClass,
-    methodName: string): Binding;
+  public server<T extends Server>(ctor: Constructor<T>, name?: string) {
+    const suffix = name || ctor.name;
+    const key = `components.${suffix}.server`;
```

Comment:
Subjectively, I don't like this naming convention - a server does not have to be bound to a component in a 1:1 way. Can we use `servers.${suffix}` instead?